app/services/nomis/booker.rb

```diff
@@ -0,0 +1,56 @@
+module Nomis
+  class Booker
+    ALREADY_PROCESSED = 'Duplicate post'.freeze
+
+    def initialize(visit)
+      self.visit = visit
+      self.api_error = false
+    end
+
+    def book
+      call_api
+      build_booking_response
+    end
+
+    def nomis_visit_id
+      booking.visit_id
+    end
+
+  private
+
+    attr_accessor :visit, :booking, :api_error
+
+    def call_api
+      self.booking = Nomis::Api.
+        instance.
+        book_visit(offender_id: offender_id, params: booking_params)
+    rescue Nomis::APIError
+      self.api_error = true
+    end
+
+    def build_booking_response
+      return BookingResponse.nomis_api_error if api_error
```

Comment:
This is related to the fact that we want to delay the API call as much as possible, so this class needs to return a `booking_response`